zwave_js/rootfs/etc/cont-init.d/config.sh

```diff
@@ -6,30 +6,42 @@ declare network_key
 
 readonly DOCS_EXAMPLE_KEY="2232666D100F795E5BB17F0A1BB7A146"
 
-if [[ "${DOCS_EXAMPLE_KEY}" == "$(bashio::config 'network_key')" ]]; then
-    bashio::log.fatal
-    bashio::log.fatal 'The add-on detected that the Z-Wave network key used'
-    bashio::log.fatal 'is from the documented example.'
-    bashio::log.fatal
-    bashio::log.fatal 'Using this key is insecure, because it is publicly'
-    bashio::log.fatal 'listed in the documentation.'
-    bashio::log.fatal
-    bashio::log.fatal 'Please check the add-on documentation on how to'
-    bashio::log.fatal 'create your own, secret, "network_key" and replace'
-    bashio::log.fatal 'the one you have configured.'
-    bashio::log.fatal
-    bashio::log.fatal 'Click on the "Documentation" tab in the Z-Wave JS'
-    bashio::log.fatal 'add-on panel for more information.'
-    bashio::log.fatal
-    bashio::exit.nok
-elif ! bashio::config.has_value 'network_key'; then
-    bashio::log.info "No Network Key is set, generate one..."
-    network_key=$(hexdump -n 16 -e '4/4 "%08X" 1 "\n"' /dev/random)
-    bashio::addon.option network_key "${network_key}"
-else
-    network_key=$(bashio::config 'network_key')
+if bashio::config.has_value 'network_key'; then
+    bashio::log.info "Migrating \"network_key\" option to \"s0_legacy_key\"..."
+    network_key=$(bashio::string.upper "$(bashio::config 'network_key')")
+    bashio::addon.option s0_legacy_key "${network_key}"
+    bashio::addon.option network_key
 fi
 
+for key in "s0_legacy_key" "s2_access_control_key" "s2_authenticated_key" "s2_unauthenticated_key"; do
+    network_key=$(bashio::jq "$(bashio::addon.options)" .${key})
```

Comment:
I guess this was done to bypass the caching issue above

```suggestion
    network_key=$(bashio::config "${key}")
```

Reply:
yes, I was not aware of the cache, thanks! I will test this locally before pushing up the changes

Reply:
Actually that will not work, let me get back to you later :)

The config is read from disk, not from remote.